qcodes/utils/dataset/doNd.py

```diff
@@ -312,7 +320,14 @@ def do1d(
     # reimplemented from scratch
     with _catch_keyboard_interrupts() as interrupted, meas.run() as datasaver:
         additional_setpoints_data = _process_params_meas(additional_setpoints)
-        for set_point in np.linspace(start, stop, num_points):
+        setpoints = np.linspace(start, stop, num_points)
+
+        # flush to prevent unflushed print's to visually interrupt tqdm bar
+        # updates
+        sys.stdout.flush()
+        sys.stderr.flush()
```

Comment:
How come we need this? I guess this still does not resolve the issue if something is printed within the dond function?

Reply:
yes, exactly. Measurement when starts it print "starting run with id: 42", and that sometimes can get flushed after tqdm started also printing. hence we flush first, and then tqdm can start printing.